File: backend/app/schemas/property.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class PropertyBase(BaseModel):
    address: str
    location: str
    price: float
    description: Optional[str] = None
    property_type: str
    square_footage: float
    year_built: int
    bedrooms: Optional[int] = None
    bathrooms: Optional[float] = None
    status: str = "available"

class PropertyCreate(PropertyBase):
    pass

class PropertyUpdate(PropertyBase):
    pass

class Property(PropertyBase):
    id: int
    owner_id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True
        json_encoders = {
            datetime: lambda dt: dt.isoformat() if dt else None
        } 

File: frontend/src/types/property.ts
export type PropertyType = 'residential' | 'commercial' | 'retail' | 'industrial' | 'land';
export type PropertyStatus = 'available' | 'pending' | 'exchanged';

export interface PropertyBase {
  address: string;
  location: string;
  price: number;
  description?: string;
  property_type: PropertyType;
  square_footage: number;
  year_built: number;
  bedrooms?: number;
  bathrooms?: number;
  status: PropertyStatus;
}

export interface PropertyCreate extends PropertyBase {}

export interface PropertyUpdate extends Partial<PropertyBase> {}

export interface Property extends PropertyBase {
  id: number;
  owner_id: number;
  created_at: string;
  updated_at: string;
}

export interface PropertyMatch {
  property: Property;
  score: number;
  value_score: number;
  location_score: number;
  type_score: number;
}

export interface ExchangeChain {
  properties: Property[];
  average_score: number;
  scores: number[];
} 